src/main/java/org/dataloader/CachedValueStore.java

```diff
@@ -0,0 +1,78 @@
+package org.dataloader;
+
+import org.dataloader.annotations.PublicSpi;
+import org.dataloader.impl.NoOpCachedValueStore;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Cache value store for data loaders that use caching and want a long-lived or external cache.
+ * <p>
+ * The default implementation is a no-op store which replies with the key always missing and doesn't
+ * store any actual results. This is to avoid duplicating the stored data between the {@link CacheMap}
+ * and the store.
+ * <p>
+ * The API signature uses completable futures because the backing implementation MAY be a remote external cache
+ * and hence exceptions may happen in retrieving values.
+ *
+ * @param <K> the type of cache keys
+ * @param <V> the type of cache values
+ *
+ * @author <a href="https://github.com/craig-day">Craig Day</a>
+ */
+@PublicSpi
+public interface CachedValueStore<K, V> {
```

Comment:
I must say I am not really convinced of the name, but I also don't anything better out of my head :-)

Reply:
What about secondLevelCache?